base/server/python/pki/server/cli/est.py

```diff
@@ -0,0 +1,169 @@
+# Copyright Red Hat, Inc.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+
+from __future__ import absolute_import
+from __future__ import print_function
+import getopt
+import logging
+import os
+import sys
+
+import pki.cli
+import pki.server
+import pki.server.instance
+import pki.server.cli.subsystem
+
+DEFAULT_SUBSYSTEM_NAME = 'est'
+DEFAULT_INSTANCE_NAME = 'pki-tomcat'
+
+logger = logging.getLogger(__name__)
+
+
+class ESTCLI(pki.cli.CLI):
+
+    def __init__(self):
+        super().__init__('est', 'Manage the EST subsystem')
+
+        self.add_module(ESTCreateCLI())
+        self.add_module(ESTRemoveCLI())
+        self.add_module(pki.server.cli.subsystem.SubsystemDeployCLI(self))
+        self.add_module(pki.server.cli.subsystem.SubsystemUndeployCLI(self))
+
+
+class ESTCreateCLI(pki.cli.CLI):
+
+    def __init__(self):
+        super().__init__('create', 'Create EST subsystem')
+
+    def print_help(self):
+        print('Usage: pki-server est-create [OPTIONS] [name]')
+        print()
+        print('  -i, --instance <instance ID>       Instance ID (default: pki-tomcat).')
+        print('      --force                        Force creation.')
+        print('  -v, --verbose                      Run in verbose mode.')
+        print('      --debug                        Run in debug mode.')
+        print('      --help                         Show help message.')
+        print()
+
+    def execute(self, argv):
+
+        try:
+            opts, args = getopt.gnu_getopt(argv, 'i:v', [
+                'instance=', 'database=', 'issuer=',
+                'force',
+                'verbose', 'debug', 'help'])
+
+        except getopt.GetoptError as e:
+            logger.error(e)
+            self.print_help()
+            sys.exit(1)
+
+        name = DEFAULT_SUBSYSTEM_NAME
+        instance_name = DEFAULT_INSTANCE_NAME
+        force = False
+
+        for o, a in opts:
+            if o in ('-i', '--instance'):
+                instance_name = a
+
+            elif o == '--force':
+                force = True
+
+            elif o in ('-v', '--verbose'):
+                logging.getLogger().setLevel(logging.INFO)
+
+            elif o == '--debug':
+                logging.getLogger().setLevel(logging.DEBUG)
+
+            elif o == '--help':
+                self.print_help()
+                sys.exit()
+
+            else:
+                logger.error('Unknown option: %s', o)
+                self.print_help()
+                sys.exit(1)
+
+        if len(args) > 0:
+            name = args[0]
+
+        instance = pki.server.instance.PKIServerFactory.create(instance_name)
+
+        if not instance.exists():
+            raise Exception('Invalid instance: %s' % instance_name)
```

Comment:
If `pki-tomcat` instance doesn't exist, this blows up:
```
# pki-server est-create
ERROR: Invalid instance: pki-tomcat
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/pki/server/pkiserver.py", line 41, in <module>
    cli.execute(sys.argv)
  File "/usr/lib/python3.10/site-packages/pki/server/cli/__init__.py", line 145, in execute
    super().execute(args)
  File "/usr/lib/python3.10/site-packages/pki/cli/__init__.py", line 217, in execute
    module.execute(module_args)
  File "/usr/lib/python3.10/site-packages/pki/cli/__init__.py", line 217, in execute
    module.execute(module_args)
  File "/usr/lib/python3.10/site-packages/pki/server/cli/est.py", line 95, in execute
    raise Exception('Invalid instance: %s' % instance_name)
Exception: Invalid instance: pki-tomcat
```

I think this should exit gracefully with non-zero code instead of throwing up an Exception, like in https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/cli/ocsp.py#L147-L149 
An exception can signify both a bug in the program and a user error. But a handled exception is more likely just a user error.
But I see both patterns used extensively in the codebase, so I'll leave this up to the team.